cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeletphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	kubeletWriteInitConfigLongDesc = normalizer.LongDesc(`
+		Write init kubelet configuration to disk for dynamic kubelet configuration feature, functionally equivalent to what implemented by kubeadm init.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletWriteInitConfigExample = normalizer.Examples(`
+		# Write init kubelet configuration to disk.
+		kubeadm alpha phase kubelet write-init-config
```

Comment:
What about adding `--config` to the examples?

Reply:
Yeah, it'd be good to give user a more clear example. I was thinking of if I should use `--config` as the flag name for this command. It may cause confusing.  `--config` is used to specify the kubeadm config. But in an example, users may think it as the config which is going to be written on the disk. @fabriziopandini any suggestions here?

Reply:
Good point. 
IMO we should be consistent across command and always use `--config` to specify the kubeadm config, which is the case also for `kubelet` subcommands; what about:
a. always use `configuration` instead of `config` for kubelet phase subcommand names. e.g.
   - `kubeadm alpha phase kubelet write-init-configuration`
   - `kubeadm alpha phase kubelet upload-dynamic-configuration`

or

b. remove `config` from kubelet phase subcommand names
  - `kubeadm alpha phase kubelet init`
  - `kubeadm alpha phase kubelet upload`

and add the following sentence (or something similar) to long description of all commands`Please note that the kubelet configuration can be passed to kubeadm as a value into the master configuration file.`

Reply:
I prefer the latter but I'm not sure if these commands are clear enough to users. Maybe we can give more tips in the description besides your suggested sentence. @luxas WDYT?

Reply:
I also prefer the second option, unless we can already think of other subcommands that may go in there. Also, it's alpha, so we can change later.

Reply:
done with the second option.